diagon-alley/src/Components/AuthForms/SignUpForm.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { Textfields, Button, Toast } from '..';
+import { SignupFormDetail } from '../../GeneralFunctions';
+import { HiEye } from 'react-icons/hi';
+import { RiEyeCloseFill } from 'react-icons/ri';
+import { emailRegex, passwordRegex } from '../../RegexForValidation/Validation';
+import axios from 'axios';
+import './Auth.css';
+import { useNavigate } from "react-router-dom";
+
+const defaultForm = {
+  firstName: "",
+  lastName: "",
+  email: "",
+  password: "",
+  confirmPassword: "",
+}
+
+const SignUpForm = () => {
+
+  let navigate = useNavigate();
+  const [form, setForm] = useState(defaultForm)
+  const [submitMode, setSubmitMode] = useState(false)
+  const [checkPassword, setCheckPassword] = useState(false)
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [error, setError] = useState({
+    email: {
+      isError: false, 
+      errorMessage: "Enter a valid mail",
+    }, password: {
+      isError: false, 
+      errorMessage: "Password must contain minimum eight characters, at least one uppercase letter, one lowercase letter and one number",
+    }, confirmPassword: {
+      isError: false,
+      errorMessage: "Password does not match",
+    }
+  })
+
+  const handleChange = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    if(name==="password") {
+      setCheckPassword(value)
+    }
+    const validateError = validateForm(name, value);
+    setError((prevValue) => ({
+      ...prevValue,
+      [name]: {
+        ...prevValue[name],
+        isError: validateError
+      }
+    }));
+    setForm({...form, [name]: value})
+  }
+  
+  const SubmitHandler = async(event) => {
+    event.preventDefault();
+    try {
+      const res = await axios.post(`/api/auth/signup`, {
+        firstName: form.firstName,
+        lastName: form.lastName,
+        email: form.email,
+        password: form.password,
+      });
+      if(res.status===201){
+        setForm(defaultForm);
+        Toast("Successfully signed up", "success")
+        //localStorage.setItem("userToken", res.data.encodedToken);
+        navigate("../login", { replace: true });
+      }
+      // saving the encodedToken in the localStorage
+      
+    } catch (error) {
+      Toast("Could not signed up, try again later.", "error")
+      console.error(error);
+    }
+  }
+
+  const validateForm = (name, value) => {
```

Comment:
The functions for signup & login can be extracted out in a file and can be used by importing them. ValidateForm is being used in both.

Reply:
yes okay, good suggestion, can reuse that too, thank you